android-client-sdk/src/main/java/com/devcycle/sdk/android/eventsource/DVCLifecycleCallbacks.kt

```diff
@@ -0,0 +1,76 @@
+package com.devcycle.sdk.android.eventsource
+
+import android.app.Activity
+import android.app.Application
+import android.os.Bundle
+import android.os.Handler
+import android.os.HandlerThread
+import com.devcycle.sdk.android.model.BucketedUserConfig
+import timber.log.Timber
+
+class DVCLifecycleCallbacks(
+    eventSource: EventSource,
+    onResume: () -> Any,
+    config: BucketedUserConfig? = null,
+    customHandler: Handler? = null
+) : Application.ActivityLifecycleCallbacks {
+    private val inactivityDelay: Long = config?.sse?.inactivityDelay?.toLong() ?: 800
+
+    private val eventSource = eventSource
+    private val onResume = onResume
+    private val listenerThread: HandlerThread = HandlerThread("DVCForegroundListener")
+    private val handler: Handler
+    private var paused = false
+    private var foreground = true
+    private var checkForegroundCallback: Runnable? = null
+
+    init {
+        if (customHandler == null) listenerThread.start()
+        handler = customHandler ?: Handler(listenerThread.looper)
+    }
+    override fun onActivityResumed(activity: Activity) {
+        paused = false
+        if (checkForegroundCallback != null) {
+            handler.removeCallbacks(checkForegroundCallback!!) // stop check from running if the timer is still running
+            checkForegroundCallback = null
+        }
+        if (!foreground) {
+            Timber.d("Restarting streaming event source connection")
+            eventSource.start()
+            onResume()
+        }
+        foreground = true
+    }
+
+    override fun onActivityPaused(activity: Activity) {
+        paused = true
+
+        if (checkForegroundCallback != null) {
+            handler.removeCallbacks(checkForegroundCallback!!)
+            checkForegroundCallback = null
+        }
+
+        checkForegroundCallback = Runnable {
+            if (foreground && paused) {
+                foreground = false
```

Comment:
it may be more accurate for this variable to be named `connectionClosed` or something, since technically the app is in the background long before this is actually set to `false`. 

it would be even better to use some property of the eventSource, if it has some kind of `.closed` property you can check

Reply:
good idea, i got it working with eventSource.state